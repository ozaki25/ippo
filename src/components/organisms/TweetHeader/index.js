import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';

const TweetHeaderTime = styled.small`
  &:before {
    content: '\00b7';
  }
`;

const TweetHeader = ({ name, time }) => (
  <>
    <strong>{name}</strong>
    <TweetHeaderTime>{time}</TweetHeaderTime>
  </>
);

TweetHeader.displayName = 'TweetHeader';

TweetHeader.propTypes = {
  name: propTypes.string.isRequired,
  time: propTypes.string.isRequired,
};

TweetHeader.defaultProps = {};

export default TweetHeader;
