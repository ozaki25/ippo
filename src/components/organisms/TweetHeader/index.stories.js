import React from 'react';
import { text } from '@storybook/addon-knobs';
import TweetHeader from '.';

export default {
  title: 'organisms/TweetHeader',
};

const props = {
  name: text('name', 'ozaki25'),
  time: text('time', '12月10日'),
};

export const 通常パターン = () => <TweetHeader {...props} />;
