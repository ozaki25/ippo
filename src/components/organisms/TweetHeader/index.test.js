import React from 'react';
import { snapshot } from 'src/test/helpers';
import TweetHeader from '.';

const props = {
  name: 'ozaki25',
  time: '12月10日',
};

snapshot('TweetHeader/nomal', <TweetHeader {...props} />);
