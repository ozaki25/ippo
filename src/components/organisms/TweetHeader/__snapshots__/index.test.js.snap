// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: TweetHeader/nomal 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MockRouter
    createHref={[Function]}
    location=""
    params={Object {}}
    path=""
    push={[Function]}
    replace={[Function]}
    url=""
  >
    <TweetHeader
      name="ozaki25"
      time="12月10日"
    >
      <strong>
        ozaki25
      </strong>
      <styled.small>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "bDWFJH",
                "rules": Array [],
              },
              "displayName": "styled.small",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "small",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <small
            className=""
          >
            12月10日
          </small>
        </StyledComponent>
      </styled.small>
    </TweetHeader>
  </MockRouter>
</Provider>
`;
